app/frontend/src/service/conversations.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  CreateGroupChatReq,
+  GetGroupChatMessagesReq,
+  GroupChat,
+  ListGroupChatsReq,
+  Message,
+  SendMessageReq,
+} from "../pb/conversations_pb";
+import client from "./client";
+import { User } from "../pb/api_pb";
+
+export async function listGroupChats(): Promise<GroupChat.AsObject[]> {
+  const req = new ListGroupChatsReq();
+
+  const response = await client.conversations.listGroupChats(req);
+  const groupChats = response.getGroupChatsList();
+
+  return groupChats.map((groupChat) => groupChat.toObject());
+}
+
+export async function getGroupChatMessages(
+  groupChatId: number
+): Promise<Message.AsObject[]> {
+  const req = new GetGroupChatMessagesReq();
+  req.setGroupChatId(groupChatId);
+
+  const response = await client.conversations.getGroupChatMessages(req);
+  const messages = response.getMessagesList();
+
+  return messages.map((message) => message.toObject());
+}
+
+export async function createGroupChat(users: User.AsObject[]): Promise<number> {
```

Comment:
I don't know if it really matters, but this and above function specify return type, but below `SendMessage` infers it.

Reply:
I like to start writing a function with the declaration and work toward the return type. sendMessage was probably too easy. But I do not mind if those declarations are not always there. This kind of consistency is not so important for me as it does not change the runtime behaviour. Although a too general typing may unnecessary make the developer write bad code. hmmmm...

Reply:
I don't think too broad typing is a big risk.